Commands System and Logging (chandler branch)

chandlerwr opened the pull request:
Addresses #15 and #14 as a possible solution.
Commands system broken into an object in its own file (prevents clutter in players.js).
Logging system also in its own file. Used via: `var logger = require("./logger.js").logger` and `logger.info("message")`.

cwr8924 committed:
Added winston

Added winston logger as logging.js (handles winston configuration).
Added Visual Studio files to .gitignore.

cwr8924 committed:
Initial command system

Added cmds.js to handle commands (prevent crowding of players.js)
Implemented logging in players.js and cmds.js
Implemented cmds.js in players.js

cwr8924 committed:
Commands system?

Implemented ttmtg command (testing command -- probably won't keep?)
Added reminders and remind command (to be implemented)
Also documentation

cwr8924 committed:
Rearranging and consolidation

Added unknownCommand for responding to unkown commands
Added sendUsage for sneding a command's usage

ben-dow commented on src/cmds.js:
```diff
@@ -0,0 +1,185 @@
+                // help with no argument
+            } else {
+
+                let msg = "Commands:"
```
Missing Semicolon

ben-dow commented on src/cmds.js:
```diff
@@ -0,0 +1,185 @@
+
+        // args should be string[], message should be the discord Message as given in the message event
+        func: function (args, message) {
+
```
As this file is likely to get very long, lets move Functionality out of this file and trigger the functionality from here. Keeping command functionality out of this file will keep it from getting absurdly long

ben-dow commented on src/cmds.js:
```diff
@@ -0,0 +1,185 @@
+        usage: config.prefix + 'ttmtg',
+        help: 'displays the time until the next RIT Players meeting',
+        func: function (args, message) {
+
```
Same issue here about Moving functionality out of this file

ben-dow commented on src/cmds.js:
```diff
@@ -0,0 +1,185 @@
+        help: 'add a reminder to this server',
+        func: function (args, message) {
+
+            logCommand(this.name, args); // logging
```
Placement of Logging in this file is good and logical

ben-dow commented on src/cmds.js:
```diff
@@ -0,0 +1,185 @@
+ * @param {Message} message the message that the command was sent in
+ */
+function unknownCommand(cmd, args, message) {
+
```
Same Comment about moving functionality. For these unknown command, sendUsage  and LogCommand may be good to make a file containing the word "utility" to indicate commands and responses that will be used, but not necessarily triggered directly by a user.

ben-dow commented on src/cmds.js:
```diff
@@ -0,0 +1,185 @@
+ * @param {Message} message the message to respond to
+ */
+function sendUsage(cmd, message) {
+    
```
If this is only being used in the Help Function, may want to move it as well into the file for functionality for the help function

chandlerwr replied:
Other functions will be using `sendUsage` if they are used incorrectly.

ben-dow reviewed (changes requested):
Comments made on individual files. Overall looks good, but to prep, for the cmd.js file to get absurdly long, should be moving functionality (where it makes sense) to separate files for readability sake

ben-dow commented:
Not sure if possible to change, but pull requests, in general, should be made to dev instead of master

cwr8924 committed:
Split up commands into packages

Changed remind to schedule.
Moved help command to util package.
Moved ttmtg command to players package.
Moved schedule command to schedule package.
Implemented the usage of the packages in the main cmds.js file.

cwr8924 committed:
T h e  s e m i c o l o n
